src/main/java/it/pagopa/transactions/services/TransactionsService.java

```diff
@@ -330,170 +344,186 @@ public Mono<RequestAuthorizationResponseDto> requestTransactionAuthorization(
-                                                                                    .status(
-                                                                                            TransactionStatusDto
-                                                                                                    .fromValue(
-                                                                                                            transactionDocument
-                                                                                                                    .getStatus()
-                                                                                                                    .toString()
-                                                                                                    )
-                                                                                    )
-                                                                    );
-                                                        }
-                                                )
-                                )
+                                                            )
+                                                            .map(this::buildTransactionInfoDto)
+                                            )
+                                    );
+                        }
                 );
 
     }
 
+    private Mono<TransactionActivated> updateTransactionAuthorizationStatus(
+                                                                            BaseTransaction transaction,
+                                                                            UpdateAuthorizationRequestDto updateAuthorizationRequestDto
+    ) {
+        UpdateAuthorizationStatusData updateAuthorizationStatusData = new UpdateAuthorizationStatusData(
+                transaction,
+                updateAuthorizationRequestDto
+        );
+
+        // FIXME Handle multiple rtpId
+        TransactionUpdateAuthorizationCommand transactionUpdateAuthorizationCommand = new TransactionUpdateAuthorizationCommand(
+                transaction.getPaymentNotices().get(0).rptId(),
+                updateAuthorizationStatusData
+        );
+
+        return transactionUpdateAuthorizationHandler
+                .handle(transactionUpdateAuthorizationCommand)
+                .doOnNext(
+                        authorizationStatusUpdatedEvent -> log.info(
+                                "Requested authorization update for rptId: {}",
+                                transaction.getPaymentNotices().get(0).rptId()
+                        )
+                )
+                .flatMap(
+                        authorizationStatusUpdatedEvent -> authorizationUpdateProjectionHandler
+                                .handle(authorizationStatusUpdatedEvent)
+                );
+    }
+
+    private Mono<it.pagopa.ecommerce.commons.documents.v1.Transaction> handleClosePayment(
```

Comment:
🚲? (not too sure)

```suggestion
    private Mono<it.pagopa.ecommerce.commons.documents.v1.Transaction> closePayment(
```

Reply:
Yes I like close payment more as method name :) 
Fixed with https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/168/commits/bc75f7b013205527a9514f97c242658da540998e